src/openvic-simulation/map/ProvinceInstance.cpp

```diff
@@ -33,7 +35,7 @@ ProvinceInstance::ProvinceInstance(
 	event_modifiers {},
 	slave { false },
 	crime { nullptr },
-	rgo { pop_type_keys },
+	rgo { new_modifier_effect_cache, pop_type_keys },
```

Comment:
Rather than having a `location_ptr` in the RGO struct which is set via function later, can't it be a province reference variable which is initialised in the RGO constructor using `*this` passed from here? That guarantees that each RGO will have its province set once exactly, no risk of it being null or set multiple times.

Reply:
No because the province pointer can be changed when its container grows and moves everything. I already tried that.

Reply:
Ah it's not when the container expands, enough space is reserved for all the province instances, instead it's because we create the `ProvinceInstance`s as r-values and move them into the container.